infra/label_evaluator_step_functions/lambdas/evaluate_currency_labels.py

```
@@ -375,15 +411,40 @@ def handler(event: dict[str, Any], _context: Any) -> dict[str, Any]:
         except Exception as e:
+            # Re-raise rate limit errors for Step Function retry
+            from receipt_agent.utils.llm_factory import AllProvidersFailedError
+            from receipt_agent.utils.ollama_rate_limit import (
+                OllamaRateLimitError,
+            )
+
+            if isinstance(e, AllProvidersFailedError):
+                logger.error(
+                    "All providers failed, propagating for Step Function retry: %s",
+                    e,
+                )
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)